crates/webidl/src/lib.rs

```diff
@@ -520,6 +522,48 @@ impl<'a> WebidlParse<&'a str> for webidl::ast::RegularAttribute {
     }
 }
 
+impl<'a> WebidlParse<&'a str> for webidl::ast::Iterable {
+    fn webidl_parse(
+        &self,
+        program: &mut backend::ast::Program,
+        first_pass: &FirstPassRecord<'_>,
+        self_name: &'a str,
+    ) -> Result<()> {
+        if util::is_chrome_only(&self.extended_attributes) {
+            return Ok(());
+        }
+
+        let throws = util::throws(&self.extended_attributes);
+        let return_value = webidl::ast::ReturnType::NonVoid(self.value_type.clone());
+        let args = [];
+        first_pass
+            .create_basic_method(
+                &args,
+                Some(&"values".to_string()),
```

Comment:
I wasn't quite sure what these two methods were doing, is this what it means to be iterable? (a `values` and an `item` method?)

Reply:
It comes from the `iterable` in webidl yeah: https://heycam.github.io/webidl/#es-iterable

I couldn't get `forEach` to work as we don't have callbacks yet.

`entries`, `keys` and `values` and some other behaviour is needed.

I messed up though `item` is a getter on the webidl which isn't supported yet.

Reply:
Thanks for this direction, it really helped me understand the codebase a lot more too!